Panic on starting

Error
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4430458]

goroutine 1 [running]:
github.com/common-fate/granted/pkg/cfaws.(*uninitCFSharedConfig).init(0x46211c0, 0xc000213078)
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/profiles.go:88 +0x18
github.com/common-fate/granted/pkg/cfaws.(*uninitCFSharedConfig).init(0xc000110690, 0xc000213078)
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/profiles.go:98 +0x8e
github.com/common-fate/granted/pkg/cfaws.GetProfilesFromDefaultSharedConfig({0x47c6130, 0xc00019a010})
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/cfaws/profiles.go:71 +0x169
github.com/common-fate/granted/pkg/assume.AssumeCommand(0xc0001a7a80)
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/assume/assume.go:28 +0x151
github.com/common-fate/granted/pkg/assume.GetCliApp.func2(0x2)
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/assume/entrypoint.go:39 +0x19
github.com/common-fate/granted/pkg/updates.WithUpdateCheck.func1(0xc0001a7a80)
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/pkg/updates/updates.go:69 +0x110
github.com/urfave/cli/v2.(*App).RunContext(0xc0002d44e0, {0x47c6130, 0xc00019a010}, {0xc00019c000, 0x2, 0x2})
        /Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:322 +0x7a8
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /Users/runner/work/granted-cli-build/granted-cli-build/granted/cmd/assume/main.go:14 +0x45
```
Version (OS as well)
```
Version: 0.1.5
Darwin xxx 20.6.0 Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64 (Big Sur)
```

managed to choose browser (Firefox), installed addon, added shell alias, any calling like `assume <profile or not>` ends up with above.

--- a/pkg/cfaws/assumer_aws_iam.go
+++ b/pkg/cfaws/assumer_aws_iam.go
@@ -59,13 +59,18 @@ func (aia *AwsIamAssumer) ProfileMatchesType(rawProfile configparser.Dict, parse
 
 // GetFederationToken is used when launching a console session with longlived IAM credentials profiles
 func getFederationToken(ctx context.Context, c *CFSharedConfig) (aws.Credentials, error) {
-	cfg := aws.NewConfig()
-	r, _, err := c.Region(ctx)
+	opts := []func(*config.LoadOptions) error{
+		// load the config profile
+		config.WithSharedConfigProfile(c.Name),
+	}
+
+	//load the creds from the credentials file
+	cfg, err := config.LoadDefaultConfig(ctx, opts...)
 	if err != nil {
 		return aws.Credentials{}, err
 	}
-	cfg.Region = r
-	client := sts.NewFromConfig(*cfg)
+
+	client := sts.NewFromConfig(cfg)
 	out, err := client.GetFederationToken(ctx, &sts.GetFederationTokenInput{Name: aws.String("Granted@" + c.Name)})
 	if err != nil {
 		return aws.Credentials{}, err

--- a/pkg/cfaws/profiles.go
+++ b/pkg/cfaws/profiles.go
@@ -58,9 +58,10 @@ func GetProfilesFromDefaultSharedConfig(ctx context.Context) (CFSharedConfigs, e
 		// Check if the section is prefixed with 'profile ' and that the profile has a name
 		if strings.HasPrefix(section, "profile ") && len(section) > 8 {
 			name := strings.TrimPrefix(section, "profile ")
-			if strings.Contains(name, ".") {
+			illegalChars := ".@"
+			if strings.ContainsAny(name, illegalChars) {
 				// The AWS SDK actually fails to parse profiles containing "." however the error it returns is not useful so we need to warn users of this
-				fmt.Fprintf(color.Error, "warning, profile: %s cannot be loaded because it contains '.' in the name, try replacing these with '-'\n", name)
+				fmt.Fprintf(color.Error, "warning, profile: %s cannot be loaded because it contains one or more of: '%s' in the name, try replacing these with '-'\n", name, illegalChars)
 				continue
 			} else {
 				cf, err := config.LoadSharedConfigProfile(ctx, name)

--- a/pkg/cfaws/profiles.go
+++ b/pkg/cfaws/profiles.go
@@ -58,7 +58,7 @@ func GetProfilesFromDefaultSharedConfig(ctx context.Context) (CFSharedConfigs, e
 		// Check if the section is prefixed with 'profile ' and that the profile has a name
 		if strings.HasPrefix(section, "profile ") && len(section) > 8 {
 			name := strings.TrimPrefix(section, "profile ")
-			illegalChars := ".@"
+			illegalChars := ".,@#$%^&*()+=\\|]}[{;:'\"<>/?"
 			if strings.ContainsAny(name, illegalChars) {
 				// The AWS SDK actually fails to parse profiles containing "." however the error it returns is not useful so we need to warn users of this
 				fmt.Fprintf(color.Error, "warning, profile: %s cannot be loaded because it contains one or more of: '%s' in the name, try replacing these with '-'\n", name, illegalChars)

--- a/pkg/cfaws/profiles.go
+++ b/pkg/cfaws/profiles.go
@@ -9,7 +9,6 @@ import (
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/bigkevmcd/go-configparser"
 	"github.com/fatih/color"
-	"github.com/pkg/errors"
 )
 
 type CFSharedConfig struct {
@@ -53,20 +52,27 @@ func GetProfilesFromDefaultSharedConfig(ctx context.Context) (CFSharedConfigs, e
 	for _, section := range configFile.Sections() {
 		rawConfig, err := configFile.Items(section)
 		if err != nil {
-			fmt.Fprintln(color.Error, errors.Wrapf(err, "failed to load a profile from your AWS config: %s . Due to the following error", section).Error())
+			fmt.Fprintf(color.Error, "failed to parse a profile from your AWS config: %s Due to the following error: %s\n", section, err)
 			continue
 		}
 		// Check if the section is prefixed with 'profile ' and that the profile has a name
 		if strings.HasPrefix(section, "profile ") && len(section) > 8 {
 			name := strings.TrimPrefix(section, "profile ")
-			cf, err := config.LoadSharedConfigProfile(ctx, name)
-
-			if err != nil {
-				fmt.Fprintln(color.Error, errors.Wrapf(err, "failed to load a profile from your AWS config: %s . Due to the following error", name).Error())
+			if strings.Contains(name, ".") {
+				// The AWS SDK actually fails to parse profiles containing "." however the error it returns is not useful so we need to warn users of this
+				fmt.Fprintf(color.Error, "warning, profile: %s cannot be loaded because it contains '.' in the name, try replacing these with '-'\n", name)
 				continue
 			} else {
-				profiles[name] = &uninitCFSharedConfig{initialised: false, CFSharedConfig: &CFSharedConfig{AWSConfig: cf, Name: name, RawConfig: rawConfig}}
+				cf, err := config.LoadSharedConfigProfile(ctx, name)
+
+				if err != nil {
+					fmt.Fprintf(color.Error, "failed to load a profile from your AWS config: %s Due to the following error: %s\n", name, err)
+					continue
+				} else {
+					profiles[name] = &uninitCFSharedConfig{initialised: false, CFSharedConfig: &CFSharedConfig{AWSConfig: cf, Name: name, RawConfig: rawConfig}}
+				}
 			}
+
 		}
 	}
 

--- a/pkg/cfaws/profiles.go
+++ b/pkg/cfaws/profiles.go
@@ -8,7 +8,6 @@ import (
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/bigkevmcd/go-configparser"
-	"github.com/common-fate/granted/pkg/debug"
 	"github.com/fatih/color"
 	"github.com/pkg/errors"
 )
@@ -54,15 +53,16 @@ func GetProfilesFromDefaultSharedConfig(ctx context.Context) (CFSharedConfigs, e
 	for _, section := range configFile.Sections() {
 		rawConfig, err := configFile.Items(section)
 		if err != nil {
-			debug.Fprintf(debug.VerbosityDebug, color.Error, "%s\n", errors.Wrap(err, "loading profiles from config").Error())
+			fmt.Fprintln(color.Error, errors.Wrapf(err, "failed to load a profile from your AWS config: %s . Due to the following error", section).Error())
 			continue
 		}
 		// Check if the section is prefixed with 'profile ' and that the profile has a name
 		if strings.HasPrefix(section, "profile ") && len(section) > 8 {
 			name := strings.TrimPrefix(section, "profile ")
 			cf, err := config.LoadSharedConfigProfile(ctx, name)
+
 			if err != nil {
-				debug.Fprintf(debug.VerbosityDebug, color.Error, "%s\n", errors.Wrap(err, "loading profiles from config").Error())
+				fmt.Fprintln(color.Error, errors.Wrapf(err, "failed to load a profile from your AWS config: %s . Due to the following error", name).Error())
 				continue
 			} else {
 				profiles[name] = &uninitCFSharedConfig{initialised: false, CFSharedConfig: &CFSharedConfig{AWSConfig: cf, Name: name, RawConfig: rawConfig}}
@@ -103,10 +103,15 @@ func (c *uninitCFSharedConfig) init(profiles map[string]*uninitCFSharedConfig, d
 					}
 				}
 			} else {
-				sourceProfile := profiles[c.AWSConfig.SourceProfileName]
-				sourceProfile.init(profiles, depth+1)
-				c.ProfileType = sourceProfile.ProfileType
-				c.Parents = append(sourceProfile.Parents, sourceProfile.CFSharedConfig)
+				sourceProfile, ok := profiles[c.AWSConfig.SourceProfileName]
+				if ok {
+					sourceProfile.init(profiles, depth+1)
+					c.ProfileType = sourceProfile.ProfileType
+					c.Parents = append(sourceProfile.Parents, sourceProfile.CFSharedConfig)
+				} else {
+					fmt.Fprintf(color.Error, "failed to load a source-profile for profile: %s . You should fix the issue with the source profile before you can assume this profile.", c.Name)
+				}
+
 			}
 		} else {
 			fmt.Fprintf(color.Error, "maximum source profile depth exceeded for profile %s\nthis indicates that you have a cyclic reference in your aws profiles.[profile dev]\nregion = ap-southeast-2\nsource_profile = prod\n\n[profile prod]\nregion = ap-southeast-2\nsource_profile = dev", c.Name)

